crates/http-api-bindings/src/fastchat.rs

```diff
@@ -0,0 +1,93 @@
+use async_trait::async_trait;
+use reqwest::header;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use tabby_inference::{TextGeneration, TextGenerationOptions};
+
+#[derive(Serialize)]
+struct Request {
+    model: String,
+    prompt: Vec<String>,
+    max_tokens: usize,
+    temperature: f32,
+}
+
+#[derive(Deserialize)]
+struct Response {
+    choices: Vec<Prediction>,
+}
+
+#[derive(Deserialize)]
+struct Prediction {
+    text: Vec<String>,
+}
+
+pub struct FastChatEngine {
+    client: reqwest::Client,
+    api_endpoint: String,
+    model_name: String,
+}
+
+impl FastChatEngine {
```

Comment:
Is this interface essentially the OpenAI interface? If so, it would be better to call it `OpenAIEngine`, and use `open-ai` in kind param.

Reply:
I think OpenAIEngine is ok for me. However fastchat has other params beside the default openAI defined param, like [use_beam_search](https://github.com/lm-sys/FastChat/blob/main/fastchat/serve/openai_api_server.py#L246).  So we  may define it as OpenAICompatibleEngine with open-ai-compatible?

Reply:
Sounds good, let's keep it as is